### PR TITLE
Fix issues found in PE-D

### LINKED ISSUE #134: HelpWindow changes
- [x] Extend Commands to be able to view the entire command and not get cut off
- [x] Fix command Labels so they are not cut off

Make HelpWindow resizable (no need, will cause more troubles)

### LINKED ISSUE #143: [PE-D] Small bug in Quick Start section: Regarding the "commands to try out"
In the Quick Start section, where there are a couple of commands listed to try out, maybe you could use more complete commands.

Because if I try the "add" command as given in the above screenshot, it (correctly) gives me an error, but this isn't ideal (i.e, going to the features to figure out the complete command) if I'm a first-time user - the purpose of this section is probably to familiarise the user with the UI, so maybe more complete commands might be nicer!

-------------
Labels: `type.DocumentationBug` `severity.Low`
original: aakanshanarain/ped#1

### LINKED ISSUE #156: [PE-D] delete 0 or delete <negative number> doesn't give specific error message
error msg not specific:

specific error msg :

The same thing also happens  for match command

-------------
Labels: `type.FeatureFlaw` `severity.Low`
original: softmagnet/ped#3

### LINKED ISSUE #158: [PE-D] Filter command given in UG doesn't work
Command: filter t q/0 1

Not sure why it gives me this error message!

-------------
Labels: `severity.High` `type.FunctionalityBug`
original: aakanshanarain/ped#6

### LINKED ISSUE #169: [PE-D] User guide can be more specific to say that `filter` continues to narrow down the search results
One might try to run

1. `match 1`
2. `filter q/0`
3. `filter q/1` (oops, changed my mind, I want a different qualification)

but that gives 0 results because 3 will filter 2's results, when one might want to filter from the full list of results from `match 1`

This behaviour is fine, but perhaps can be better conveyed in the UG

-------------
Labels: `severity.Medium` `type.DocumentationBug`
original: xlzior/ped#7

### PR DESCRIPTION
**Pull request by Lemonsr**
What's done in this PR:

- [x]  Reformatted s/t to t/s for `delete` (contributes to fixing #135)
- [x]  Redesigned Help Window and prevented cutting off command texts (fixes #134)
- [x]  Updated UG to remove `t` from `filter` command specifications
- [x]  Updated UG to add clarification to `filter` command (fixes #169)
- [x]  Updated naming conventions for `CliTutors` in DG (contributes to fixing #139)
- [x]  Updated UG to fix typo in `filter` example (fixes #158, #172, #175, #176)
- [x]  Updated UG's Quick Start instructions to be clearer for new users (fixes #143)
- [x]  Improved error message for invalid indices (fixes #156, #167)

Remaining Issues:

- [ ]  Add character limit (100 characters) to `Name` and `Remark` fields and wrap text (for #161, #164)

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -273,23 +284,30 @@ Example:<br>
...
-  e.g. `filter t n/John n/Amy` will find names that match `Amy`.
+  e.g. `filter n/John n/Amy` will find names that match `Amy`.
 - Only full words will be matched. <br>
   e.g. `n/Han` will not match `Hans`, and `n/Han Solo` will not match `Han Dan`.
 - **At least one** of the optional fields must be provided.
 
+<div markdown="block" class="alert alert-warning">
+<b>:heavy_exclamation_mark: Caution:</b> <br> 
+
+* The use of `filter` should be preceded by a `match`. i.e. `filter` cannot be used on an empty Match List.
```
Comment (zihaooo9) [inline docs/UserGuide.md:301]: i dont think this makes sense cause the user can do a `filter q/0` and then `filter g/F` without having to do a  `match` in between

File: docs/UserGuide.md [hunk 2]
```diff
@@ -16,11 +16,13 @@ nav-text: User Guide
...
-| Syntax                                                                  | Description                                         |
-| ----------------------------------------------------------------------- | --------------------------------------------------- |
-| `Codeblock`                                                              | Denotes distinct classes, their methods or examples |
-| <div markdown="span" class="alert alert-info">:information_source: Note</div> | Useful information/tips                    |
-| <div markdown="span" class="alert alert-warning">:heavy_exclamation_mark: Caution</div> | Important things to take note of                    |
+| Syntax              | Description                                         |
+| ------------------- | --------------------------------------------------- |
+| **Bold**            | Important words to note                             |
+| `Codeblock`         | Denotes distinct classes, their methods or examples |
+| <kbd>Keyboard</kbd> | Keyboard actions                                    |
```
[the remark this answers is not in the document]
Comment (Lemonsr) [reply docs/UserGuide.md:23]: I've added the `Enter` keyboard action in the Quick start section now!

File: docs/UserGuide.md (hunk 1 above)
Comment (Lemonsr) [reply docs/UserGuide.md:301]: Okay, maybe I can just state that "`filter` cannot be used on an empty Match List."?

File: docs/UserGuide.md [hunk 3]
```diff
@@ -29,15 +31,24 @@ Before going into the rest of the content in our user guide, here are a few **im
...
-    - `list`: lists all tutors/students
-    - `help`: shows all commands available
-    - `exit`: exits the program
-6. Refer to the [Features](#features) below for details of each command.
+   
+2. [Download](https://github.com/AY2122S1-CS2103T-T17-2/tp/releases/tag/v1.3) the latest `clitutors.jar`.
+   
+3. Copy the file to the folder you want to use as the *home folder* for `CliTutors`.
+   
+4. **Double-click** the file to start the application. A GUI containing the annotated **5 main components** should show up as below. Note how the application contains some sample data: <br> <img src="images/UiLabelled.png" alt="Labelled Ui" width="1200"/>
```
[the remark this answers is not in the document]
Comment (Lemonsr) [reply docs/UserGuide.md:39]: I'll change after seeing if the size is too small on the actual UG!

[the remark this answers is not in the document]
Comment (zihaooo9) [reply docs/UserGuide.md:39]: sure thing!

File: docs/UserGuide.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (zihaooo9) [reply docs/UserGuide.md:23]: OH OK think i missed it opps

File: docs/UserGuide.md (hunk 1 above)
Comment (zihaooo9) [reply docs/UserGuide.md:301]: sounds good!

**Review by knotstoks — APPROVED** [review]
LGTM! 🚀

**Pull request merged**